File: src/main/scala/info/crlog/higgs/protocol/BosonMessage.scala
package info.crlog.higgs.protocol

import info.crlog.higgs.util.StringUtilities
import info.crlog.higgs.{ByteConvertible, Message}

/**
 * Courtney Robinson <courtney@crlog.info>
 */
abstract class BosonMessage[K <: ByteConvertible, V <: ByteConvertible] extends Message[K, V] {
  protected var properties = scala.collection.mutable.Map.empty[K, V]
  private var util = new StringUtilities
}

File: higgs-boson/src/main/java/com/fillta/higgs/boson/BosonMessage.java
package com.fillta.higgs.boson;

/**
 * @author Courtney Robinson <courtney@crlog.info>
 */
public class BosonMessage {
    public Object[] arguments;
    public String method;
    public String callback;
    public short protocolVersion = 0x1;

    public BosonMessage(Object[] arguments, String method, String callback, short protocolVersion) {
        this.arguments = arguments;
        this.method = method;
        this.callback = callback;
        this.protocolVersion = protocolVersion;
    }

    public BosonMessage() {

    }
}
